opts/network.go

```diff
@@ -0,0 +1,77 @@
+package opts
+
+import (
+	"encoding/csv"
+	"fmt"
+	"github.com/docker/docker/api/types/swarm"
+	"strings"
+)
+
+const (
+	networkOptName  = "name"
+	networkOptAlias = "alias"
+	networkOptIP    = "ip"
+	networkOptIPv6  = "ipv6"
+)
+
+// NetworkOpt represents a network config in swarm mode.
+type NetworkOpt struct {
+	networkAttachments []swarm.NetworkAttachmentConfig
+}
+
+// Set networkopts value
+func (n *NetworkOpt) Set(value string) error {
+	csvReader := csv.NewReader(strings.NewReader(value))
+	fields, err := csvReader.Read()
+	if err != nil {
+		return err
+	}
+	var netAttach swarm.NetworkAttachmentConfig
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		//Support legacy non csv format
+		if len(parts) == 1 {
+			netAttach.Target = parts[0]
+			break
+		}
+
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid field %s", field)
+		}
+
+		key := strings.TrimSpace(strings.ToLower(parts[0]))
+		value := strings.TrimSpace(strings.ToLower(parts[1]))
+
+		switch key {
+		case networkOptName:
+			netAttach.Target = value
+		case networkOptAlias:
+			netAttach.Aliases = append(netAttach.Aliases, value)
+		default:
+			return fmt.Errorf("invalid field key %s", key)
```

Comment:
I think we should make the error messages consistent with what we have for `--mount`, e.g. https://github.com/moby/moby/blob/49376cddab7d2ea28fd08fbbb75522656a72ed09/opts/mount.go#L129